Robinhood/Robinhood.py

```diff
@@ -1282,6 +1431,49 @@ def submit_order(self,
                 payload[field] = value
 
         res = self.session.post(self.endpoints['orders'], data=payload)
-        res.raise_for_status()
+        self.raiseForStatus(res)
+
+        return res
+
+    ##############################
+    # CANCEL ORDER
+    ##############################
+    def cancel_order(self,oid):
+        """
+        Cancel a given order id
+
+        Args:
+            oid (string): the order ID to be cancelled
+        Returns:
+            (:obj:`requests.request`): result from `orders` post command
+            status code 200 signifies success
 
+        """
+        data = {}
+        data['oid'] = oid
+        res = self.session.post(self.endpoints['cancel'].format(**data))
+        self.raiseForStatus(res)
         return res
+
+
+    def cancel_all_orders(self,instrument=None):
+        """
+        Convenience function to cancel all orders, optionally only for a given instrument
+
+        Args:
+            instrument (dict|str): valid instrument representation
+
+        Returns:
+            (:obj:`dict`): containing keys 'cancelled' or 'error' with the list of order ids
+        """
+        orders = self.order_history(instrument=instrument)
+        res = {'cancelled':[], 'error':[]}
+
+        for order in orders['results']:
+            if not self.OrderState(order['state']).stopped():
```

Comment:
another complicated block prone to error.  

Shouldn't `raise_for_status` be enough to catch bad cancels?  Should the loop be moved out to a function level so something like [retry](https://pypi.python.org/pypi/retry/) can be added?

Reply:
No, cancel_all_orders is checking the status from the orders returned from the RH API.  As far as I can see the API has no way of querying orders by state, so you don't really want to waste time cancelling all orders from all time if they're not active.  

As to the return, you are right that the raise_for_status in cancel_order() now comes into play, and there should be a try: block around cancelling.  The intent was that the returned data is custom here because it tries to cancel all of the open ones and provides final state at the end.